### PR TITLE
Upgrade create-react-app to v5

### PR DESCRIPTION
**Pull request by prymitive**
Some other dependencies needs to be upgraded as well, plus some typescript errors fixed.

Signed-off-by: Łukasz Mierzwa <l.mierzwa@gmail.com>

### CODE REVIEW

**Comment by prymitive** [comment]
I couldn't build UI on my laptop because build fails due to me using too recent nodejs (v16), so I've decided to give it a shot with cra upgrade to v5, which makes it build correctly.

**Comment by juliusv** [comment]
Thanks! The deploy preview at https://deploy-preview-10396--prometheus-react.netlify.app/ shows an empty page and a JS error in the PromQL grammar code:

```
promql.terms.ts:555 Uncaught TypeError: (0 , r.snippet) is not a function
    at Object.1068 (promql.terms.ts:555:12)
    at n (bootstrap:19:32)
    at Object.1624 (hybrid.ts:65:1)
    at n (bootstrap:19:32)
    at Object.8195 (index.ts:14:1)
    at n (bootstrap:19:32)
    at promql.ts:62:1
    at index.tsx:26:111
    at index.tsx:26:111
```

Does that happen for you too?

**Comment by prymitive** [comment]
Yes, trying to debug

**Comment by prymitive** [comment]
`Uncaught TypeError: autocomplete_1.snippet is not a function` which seems to point at

```
}];
exports.snippets = [{
  label: 'sum(rate(__input_vector__[5m]))',
  type: 'function',
  detail: 'snippet',
  info: 'Sum over rates of increase',
  apply: (0, autocomplete_1.snippet)('sum(rate(${__input_vector__}[5m]))')
}, {
```

Probably some issue with ESM due to deps upgrades

**Comment by prymitive** [comment]
the problem is triggered by importing commonjs files here:

https://github.com/prometheus/prometheus/blob/a4ad2909878a4aad20ddb9e24425cdf8e565008b/web/ui/react-app/src/pages/graph/ExpressionInput.tsx#L23

If I change this import to ESM it work fine

File: web/ui/react-app/package.json [hunk 1]
```diff
@@ -16,9 +16,9 @@
     "@codemirror/state": "^0.19.6",
     "@codemirror/view": "^0.19.27",
     "@forevolve/bootstrap-dark": "^1.0.0",
-    "@fortawesome/fontawesome-svg-core": "^1.2.36",
-    "@fortawesome/free-solid-svg-icons": "^5.7.2",
-    "@fortawesome/react-fontawesome": "^0.1.16",
+    "@fortawesome/fontawesome-svg-core": "1.2.36",
+    "@fortawesome/free-solid-svg-icons": "5.15.4",
+    "@fortawesome/react-fontawesome": "0.1.16",
```
Comment (prymitive) [inline web/ui/react-app/package.json:21]: pinning this release since 0.1.17 will cause build failures as it works with FA 6.0

Comment (prymitive) [reply web/ui/react-app/package.json:21]: I see that we can upgrade it to 6.0 easily, some icons will look a bit different but no breaking changes, especially that there aren't that many icons used anyway

Comment (juliusv) [reply web/ui/react-app/package.json:21]: Yeah, sounds good to me if you want to try upgrading to 6.0.

**Comment by juliusv** [comment]
Looks good to me, but will wait for the font-awesome update before merging, if you still want to try that. Thanks! :+1:

**Comment by juliusv** [comment]
:+1: Awesome, thanks!

**Pull request merged**